xapian-core/weight/tfidfweight.cc

```diff
@@ -89,6 +89,9 @@ TfIdfWeight::TfIdfWeight(const std::string &normals, double slope, double delta)
 	need_stat(DOC_LENGTH_MAX);
 	need_stat(UNIQUE_TERMS);
     }
+    if (!strchr("ntpfsP", normalizations[1])) {
+	need_stat(COLLECTION_FREQ);
+    }
```

Comment:
Perhaps better to check against a list of those normalisations which do need it, rather than those than don't?  Especially as that's a shorter list.

Reply:
Right, I will take of it .